core/tests/protractor_utils/editor.js

```diff
@@ -227,6 +228,152 @@ var expectCannotDeleteInteraction = function() {
+      null, customizationArgs);
+  }
+
+  element(by.css('.protractor-test-save-gadget-button')).click();
+
+  // Wait for the customization modal to close.
+  general.waitForSystem();
+
+};
+
+// currentName must exist.
+var renameGadget = function(currentName, newName) {
+  element(by.css('.protractor-test-rename-' + currentName + '-gadget-icon'))
+    .click();
+  var gadgetNameInput = element(by.model('newNameForRenamedGadget'));
+  gadgetNameInput.clear().then(function() {
+    gadgetNameInput.sendKeys(newName);
+  });
+  element(
+    by.css('.protractor-test-gadget-rename-confirmation-button')).click();
+};
+
+// gadgetName must exist.
+var deleteGadget = function(gadgetName) {
+  element(by.css('.protractor-test-delete-' + gadgetName + '-gadget-icon'))
+    .click();
+  general.waitForSystem(); // wait for modal popup.
+  element(by.cssContainingText('.btn-danger', 'Delete Gadget')).click();
+};
+
+// Opens the editor modal for the specified gadget.
+// This method is a precursor to other methods like adjusting visibility.
+var openGadgetEditorModal = function(gadgetName) {
+  element(by.css('.protractor-test-edit-' + gadgetName + '-gadget')).click();
+};
+
+var saveAndCloseGadgetEditorModal = function() {
+  element(by.css('.protractor-test-save-gadget-button')).click();
+};
+
+// Check (activate) visibility for a given state.
+// openGadgetEditorModal must be called before this method.
+var checkGadgetVisibilityForState = function(stateName) {
+  element(by.cssContainingText(
+    '.protractor-test-state-visibility-checkbox-label',
+    stateName)).all(by.tagName('input')).then(function(items) {
+      items[0].isSelected().then(function(selected) {
+        if (!selected) {
+          items[0].click();
+        }
+      })
+    })
+};
+
+// Uncheck (disable) visibility for a given state.
+// openGadgetEditorModal must be called before this method.
+var uncheckGadgetVisibilityForState = function(stateName) {
+  element(by.cssContainingText(
+    '.protractor-test-state-visibility-checkbox-label',
+    stateName)).all(by.tagName('input')).then(function(items) {
+      items[0].isSelected().then(function(selected) {
+        if (selected) {
+          items[0].click();
+        }
+      })
+    })
+};
+
+// This can receive additional arguments for the gadget's customizationArgs.
+// Keep in mind this method applies to a gadget preview which may not be
+// fully operational.
+var expectGadgetPreviewToMatch = function(gadgetType, gadgetName) {
+  // Convert additional arguments to an array to send on.
+  var args = [element(
+    by.css('.protractor-test-' + gadgetName + '-gadget'))];
+  for (var i = 2; i < arguments.length; i++) {
+    args.push(arguments[i]);
+  }
+  gadgets.getGadget(gadgetType).
+    expectGadgetPreviewDetailsToMatch.apply(null, args);
+};
+
+var expectGadgetWithNameDoesNotExist = function(gadgetName) {
+  element.all(by.css('.protractor-test-' + gadgetName + '-gadget'))
```

Comment:
You can use:
expect(element.all(.....).count()).toBe(0) here; count() returns a promise of the length of the array.